src/controllers/donorController.ts

```diff
@@ -61,21 +65,99 @@ export const getDonations = (req: Request, res: Response) => {
  * Posts Donations
  * @route POST /donations
  */
-export const postDonations = (req: Request, res: Response) => {
-    const donation = new Donation(req.body);
-    return donation.save()
-    .then(result => {
-        return res.status(201).json({
-            donation: result
-        });
-    })
-    .catch(error => {
-        return res.status(500).json({
-            message: error.message
-        });
-    });
+export const postDonations = async (req: Request, res: Response) => {
+    const jsonBody = JSON.parse(req.body.json);
+    try {
+        if (!req.files.descriptionImage && !jsonBody.description) {
```

Comment:
Right now `description` in Donation.ts is a mandatory field; we should make it optional.